src/org/zaproxy/zap/extension/domxss/TestDomXSS.java

```diff
@@ -574,5 +584,75 @@ public int getWascId() {
         		AlertThreshold.MEDIUM};
     }
 
+	@Override
+	public void scan(HttpMessage msg, NameValuePair originalParam) {
+		if (originalParam.getType() != NameValuePair.TYPE_QUERY_STRING) {
+			return; // Exit if it isn't a URL param
+		}
+		super.scan(msg, originalParam);
+	}
+
+	@Override
+	public void scan(HttpMessage msg, String paramName, String attack) {
+		Stats.incCounter("domxss.scan.count");
+
+		ArrayList<WebDriverWrapper> drivers = new ArrayList<WebDriverWrapper>();
+
+		WebDriverWrapper fxDriver;
+		try {
+			fxDriver = this.getFirefoxDriver();
+		} catch (Exception e) {
+			getLog().warn("Skipping scanner, failed to start Firefox: " + e.getMessage());
+			getParent().pluginSkipped(this, Constant.messages.getString("domxss.skipped.reason.browsererror"));
+			return;
+		}
+
+		drivers.add(fxDriver);
+
+		try {
+			for (String attackVector : PARAM_ATTACK_STRINGS) {
+				if (scan(drivers, paramName, attackVector)) {
+					if (!Plugin.AlertThreshold.LOW.equals(this.getAlertThreshold())) {
+						// Only report one issue per URL unless
+						// Alert threshold is LOW
+						break;
+					}
+				}
+			}
+		} finally {
+			this.returnFirefoxDriver(fxDriver);
+		}
+	}
+	
+	public boolean scan(ArrayList<WebDriverWrapper> drivers, String paramName, String attackVector) {
+		HttpMessage msg = getBaseMsg();
+		TreeSet<HtmlParameter> urlParams = msg.getUrlParams();
+		for (HtmlParameter param : urlParams) {
+			if (param.getName().equals(paramName)) {
+				param.setValue(attackVector);
+			}
+		}
+		msg.setGetParams(urlParams);
```

Comment:
It could use `setEscapedParameter` (if the attacks are already escaped as needed), otherwise worth adding a comment explaining why not use that.